### PR TITLE
[ci] Added GitHub workflow to replicate commits to version branch

### PR DESCRIPTION
**Pull request by pandafy**
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [ ] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

### CODE REVIEW

**Comment by pandafy** [comment]
I have tested the GitHub action on my fork: 

- [Creating a new branch from the version number](https://github.com/pandafy/openwisp-utils/actions/runs/12392581882) 
- [Replicating the master branch](https://github.com/pandafy/openwisp-utils/actions/runs/12392528667)

File: .github/workflows/version-branch.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+
+    - name: Get version
+      id: get_version
+      run: |
+        VERSION=$(python -c '
+        from openwisp_utils import VERSION
+        print(f"{VERSION[0]}.{VERSION[1]}.x")
```
Comment (nemesifier) [inline .github/workflows/version-branch.yml:30]: Wouldn't this produce something like `1.2.x`? We are using branch names without the x now.

File: .github/workflows/version-branch.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Rebase changes onto version branch
+      run: |
+        if git ls-remote --heads origin $VERSION | grep -sw $VERSION; then
+          git fetch origin --unshallow
+          git checkout -b $VERSION origin/$VERSION
+          git rebase origin/master
+        else
+          git checkout -b $VERSION
+        fi
+        git push origin $VERSION
```
Comment (nemesifier) [inline .github/workflows/version-branch.yml:48]: What happens here if we ever need to push force on master?

File: openwisp_utils/__init__.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-VERSION = (1, 1, 1, 'final')
+VERSION = (1, 2, 1, 'alpha')
```
Comment (nemesifier) [inline openwisp_utils/__init__.py:1]: why 1.2.1?

**Review by nemesifier — CHANGES_REQUESTED** [review]
Is there a way to create a reusable github action so we don't need to duplicate this in each repo? That would allow us to change this for all repos from a single place.

**Comment by pandafy** [comment]
> Is there a way to create a reusable github action so we don't need to duplicate this in each repo? That would allow us to change this for all repos from a single place.

We will need to find a way to automatically get the module name. Maybe, we can set this as an input parameter in GitHub workflow.

**Review by nemesifier — APPROVED** [review]

**Pull request merged**